lib/src/results.dart

```diff
@@ -94,6 +82,35 @@ class RealmResults<T extends RealmObjectBase> extends collection.IterableBase<T>
   }
 }
 
+// The query operations on results, as well as the ability to subscribe for notifications,
+// only work for results of objects (core restriction), so we add these as an extension methods
+// to allow the compiler to prevent misuse.
+extension RealmResultsOfObject<T extends RealmObjectBase> on RealmResults<T> {
+  RealmResults<T> snapshot() {
+    final handle = realmCore.resultsSnapshot(this);
+    return RealmResults<T>._(handle, realm, _metadata);
+  }
+
+  /// Returns a new [RealmResults] filtered according to the provided query.
+  ///
+  /// The Realm Dart and Realm Flutter SDKs supports querying based on a language inspired by [NSPredicate](https://academy.realm.io/posts/nspredicate-cheatsheet/)
+  /// and [Predicate Programming Guide.](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/Predicates/AdditionalChapters/Introduction.html#//apple_ref/doc/uid/TP40001789)
```

Comment:
This should link to https://www.mongodb.com/docs/realm/realm-query-language/ not the incorrect cheatsheet.